### PR TITLE
News and compatibility annotations for Julia 1.1

### PR DESCRIPTION
**Pull request by fredrikekre**
Me and @mortenpi have gone through the ~570 PRs that have been merged since Julia 1.0 (https://docs.google.com/spreadsheets/d/1PThPlhFwlkcmmEpe8JcJ-XyTdU-I1lmEh7iCw8AvFSM/edit?usp=sharing20) in preparation for the 1.1 release. We have added news and compatibility annotations for changes worthy of that.

Since we now have one doc for all of Julia 1.X (#26825) we have thus added small `!!! compat` notes to the manual for e.g. new functions and new behaviors. For a taste of what these annotations look like see e.g. https://julialang.github.io/Pkg.jl/dev/registries/

Walking through the PRs like this is not very efficient (and pretrty boring), so for future features I ask people to add these annotations together with NEWS.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
OMG, you two are saints!

**Comment by mortenpi** [comment]
> so for future features I ask people to add these annotations together with NEWS.

Should we add a note to [CONTRIBUTING.md](https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#news-worthy-changes)? Something along the lines of:

````
#### Annotations for new features, deprecations and behaviour changes

API additions and deprecations, and minor behavior changes are allowed in minor version releases. For documented features that are part of the public API, a compatibility note should be added into the manual or the docstring. It should state the Julia minor version that made changed the behavior and have a brief message describing the change.

At the moment, this should always be done with the following `compat` admonition (so that it would be possible to find the compat annotations programmatically in the future):

```
!!! compat "Julia 1.X"
    This method was added in Julia 1.X.
```
````

**Comment by JeffBezanson** [comment]
This is amazing. I was just starting to worry about how we were going to do this for this release and who was going to do it (me?) but now here it is :)

**Comment by StefanKarpinski** [comment]
I cannot emphasize how wonderful it is that you two did this. Thank you sooo much! ❤️❤️❤️

File: NEWS.md [hunk 1]
```diff
@@ -59,10 +61,41 @@ Standard library changes
...
+    `--project=@.` ([#29108]).
+  * Exponentiation operator `^` now supports raising a `Number` to an `AbstractMatrix` power ([#29782]).
 
 Compiler/Runtime improvements
 -----------------------------
 
+External dependencies
+---------------------
+
+  * 7zip (bundled with Julia on Windows) have been upgraded from version 16.04 to 18.05 ([#30035]).
```
Comment (ararslan) [inline NEWS.md]: "Have" should be "has" here and for OpenBLAS and LLVM.

Comment (fredrikekre) [reply NEWS.md]: :sweden:

File: base/multidimensional.jl [hunk 2]
```diff
@@ -211,6 +211,28 @@ module IteratorsMD
     CartesianIndex(1, 2)
     ```
 
+    ## Broadcasting
+
+    `CartesianIndices` support broadcasting arithmetic (+ and -) with a `CartesianIndex`.
+
+    !!! compat "Julia 1.1"
+    Broadcasting of CartesianIndices requires at least Julia 1.1.
```
Comment (mbauman) [inline base/multidimensional.jl]: Does this need to be indented?

**Comment by mbauman** [comment]
It's not a very strong preference — which is why I figured I'd raise the possibility on slack instead of here.  My thought is just to differentiate things a bit more from warnings, but the green does a pretty good job of that.  And we also use the encircled ! for `!!! note`s, too, so let's not worry about the color of this 🚲🏠 too much.

**Pull request merged**